torchmultimodal/modules/encoders/mdetr_text_encoder.py

```diff
@@ -0,0 +1,284 @@
+            dropout (float): dropout value in each layer. Default: 0.1.
+            normalize_before (bool): Whether to do PreNorm in encoder layers.
+                Default: False
+            return_all_layers (bool) Whether to return all layers (or just the last
+                one). Default: False
+
+    Inputs: embeddings (Tensor): Tensor of embeddings of a batch of input IDs.
+            attention_mask (Optional[Tensor]) Batch attention mask returned from
+                tokenizer (applied as padding mask inside self-attention).
+                Default: None
+    """ ""
+
+    def __init__(
+        self,
+        embedding_dim: int,
+        num_encoder_layers: int,
+        num_attention_heads: int,
+        ffn_dimension: Optional[int] = None,
+        dropout: float = 0.1,
+        normalize_before: bool = False,
+        return_all_layers: bool = False,
+    ):
+        super().__init__()
+        ffn_dimension = ffn_dimension or 4 * embedding_dim
+        layer = torch.nn.TransformerEncoderLayer(
+            d_model=embedding_dim,
+            nhead=num_attention_heads,
+            dim_feedforward=ffn_dimension,
+            dropout=dropout,
+            activation="gelu",
+            batch_first=True,
+            norm_first=normalize_before,
+        )
+        self.layers = torch.nn.TransformerEncoder(
+            encoder_layer=layer, num_layers=num_encoder_layers
+        )
+        self.normalize_before = normalize_before
+        self.embedding_layer_norm = nn.LayerNorm(embedding_dim)
+        self.return_all_layers = return_all_layers
+        self.embedding_dim = embedding_dim
+
+    def _forward_return_all_layers(
+        self,
+        embeddings: Tensor,
+        attention_mask: Optional[Tensor] = None,
+    ) -> List[torch.Tensor]:
+        encoded = embeddings
+        states = [encoded]
+        for layer in self.layers.layers:
+            encoded = layer(encoded, src_key_padding_mask=attention_mask)
+            states.append(encoded)
+        if self.normalize_before:
+            for i, state in enumerate(states):
+                states[i] = self.embedding_layer_norm(state)
+        return states
+
+    def _forward_return_last_layer(
+        self,
+        embeddings: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        encoded = embeddings
+        for layer in self.layers.layers:
+            encoded = layer(encoded, src_key_padding_mask=attention_mask)
+        if self.normalize_before:
+            encoded = self.embedding_layer_norm(encoded)
+        return encoded
+
+    def forward(
+        self,
+        embeddings: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        out: Union[torch.Tensor, List[torch.Tensor]]
+        if self.return_all_layers:
+            out = self._forward_return_all_layers(embeddings, attention_mask)
```

Comment:
do u even need this? doesnt mdetr use last one only?

Reply:
Yeah it is more for consistency with TorchText. But lemme remove